crates/forge/src/lib.rs

```diff
@@ -269,10 +287,37 @@ fn run_tests_from_file(
 
     let mut results = vec![];
     for (i, case) in tests.test_cases.iter().enumerate() {
-        let result = run_from_test_case(&runner, case, contracts, predeployed_contracts)?;
+        let case_name = case.name.as_str();
+        let function = runner.find_function(case_name)?;
+        let args = args_for_function(function);
+
+        let result = if args.is_empty() {
```

Comment:
It is very poorly abstracted here, it will be a hell to parallelize here (which we plan to do very soon i suppose). It doesn't need to be done in this PR.
Maybe instead we should create a lazy scheduler which will create a list of tuples (function, args) , for normal functions it will be an single element array. 

```
trait TestScheduler {
   fn get_schedule(function) -> Vec<ScheduledTestCase>;
}
```
and store a map of those schedules which will extract their cases from the resulting list. This way it would be trivially parallelizable. On the other hand we should consider that threads running cased for the same function have to share the state. Just leaving as a heads up.

Reply:
Will be done with #662 to make changes related to parallelism in one PR